File: event-service/src/main/java/com/tickets/event_service/event/Event.java
package com.tickets.event_service.event;

import com.tickets.event_service.category.Category;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "events")
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    // ID del organizador — viene del auth/user service, no hay FK cruzada
    @Column(name = "organizer_id", nullable = false)
    private UUID organizerId;

    @Column(length = 255, nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @Column(length = 255, nullable = false)
    private String venue;

    @Column(length = 100, nullable = false)
    private String city;

    @Column(length = 100, nullable = false)
    private String country;

    @Column(name = "start_date", nullable = false)
    private LocalDateTime startDate;

    @Column(name = "end_date")
    private LocalDateTime endDate;

    @Column(name = "image_url", length = 500)
    private String imageUrl;

    @Enumerated(EnumType.STRING)
    @Column(length = 50, nullable = false)
    private EventStatus status = EventStatus.DRAFT;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/eventFlow/src/features/events/domain/entities/Event.ts
import type { TicketType } from './TicketType'
import type { Category } from './Category'

export type EventStatus = 'DRAFT' | 'PUBLISHED' | 'CANCELLED'

export interface Event {
  id: string
  organizerId: string
  title: string
  description: string
  category: Category | null
  venue: string
  city: string
  country: string
  startDate: string
  endDate: string
  imageUrl: string | null
  status: EventStatus
  createdAt: string
  updatedAt: string
}

export interface EventWithTicketTypes extends Event {
  ticketTypes: TicketType[]
}

export interface CreateEventRequest {
  title: string
  description?: string
  categoryId?: number
  venue: string
  city: string
  country: string
  startDate: string
  endDate?: string
  imageUrl?: string
}

export interface UpdateEventRequest {
  title?: string
  description?: string
  categoryId?: number
  venue?: string
  city?: string
  country?: string
  startDate?: string
  endDate?: string
  imageUrl?: string
}
